lib/ansible/parsing/vault/__init__.py

```diff
@@ -380,39 +379,42 @@ def _shred_file(self, tmp_path):
             # we could not successfully execute unix shred; therefore, do custom shred.
             self._shred_file_custom(tmp_path)
 
-        os.remove(tmp_path)
-
     def _edit_file_helper(self, filename, existing_data=None, force_save=False):
 
         # Create a tempfile
         _, tmp_path = tempfile.mkstemp()
 
-        if existing_data:
-            self.write_data(existing_data, tmp_path, shred=False)
+        def edit_tmp_file(tmp_file_path):
+            if existing_data:
+                self.write_data(existing_data, tmp_file_path)
+
+            # drop the user into an editor on the tmp file
+            call(self._editor_shell_command(tmp_file_path))
+
+            b_tmpdata = self.read_data(tmp_file_path)
+
+            return b_tmpdata
 
-        # drop the user into an editor on the tmp file
         try:
-            call(self._editor_shell_command(tmp_path))
-        except:
-            # whatever happens, destroy the decrypted file
+            b_tmpdata = edit_tmp_file(tmp_path)
+        except Exception as e:
+            print(e)
```

Comment:
Think I was intending to remove that except stanza entirely and forgot to remove that debugging stanza